pkg/xds/envoy/names/resource_names.go

```diff
@@ -65,6 +65,14 @@ func GetMetricsHijackerClusterName() string {
 	return Join("kuma", "metrics", "hijacker")
 }
 
+func GetOpenTelemetryListenerName(backendName string) string {
```

Comment:
Again I am not sure about naming of resources, adding `listener` and `cluster` to names feel weird, but they need to be unique

Reply:
we can't have a cluster and a listener with the same name?